src/metabase/db/encryption.clj

```diff
@@ -0,0 +1,48 @@
+(ns metabase.db.encryption
+  (:require
+   [metabase.util.encryption :as encryption]
+   [metabase.util.honey-sql-2 :as h2x]
+   [metabase.util.i18n :refer [trs]]
+   [metabase.util.json :as json]
+   [toucan2.core :as t2]))
+
+(set! *warn-on-reflection* true)
+
+(defn encrypt-db
+  "Encrypt the db using the current `MB_ENCRYPTION_SECRET_KEY` to read data, and the passed `to-key` to re-encrypt.
+  If passed to-key is nil, it encrypts with the current MB_ENCRYPTION_SECRET_KEY value.
+  If passed to-key is empty string, it decrypts the entire database"
```

Comment:
The `to-key` arg that defines these behaviors isn’t immediately intuitive or explicit. Maybe this could use some more concrete strategy identifiers instead of `""` and `nil`. So, instead of relying on implicit mappings the `to-key` arg could be something like a string, `:current`, or `:decrypt`? That would make lines 18 and 19 read better too.

Reply:
or maybe we should have a `decrypt-mb` function, I'm a little unsure about having `encrypt-db` Decrypt the db.

Reply:
The "" vs nil was something I wasn't sure about either. And true with the "encrypt-db" also decrypt the db. 

The trouble is that in the end I think we need the same logic to modify all the values to either encrypted or decrypted and it ends up being the same code, since the actual encryption/decryption is handled in the passed function.

Maybe I should make what's currently `encrypt-db` into a private function that takes the `make-encrypt-fn` as an argument and then an `encrypt-db` and `decrypt-db` function which really creates the correct function to pass along? That should avoid the nil vs "" thing and make for a better external API.

Reply:
I updated the code to work that way, what do you think of it now?